python/dgl/nn/pytorch/conv/gatedgcnconv.py

```diff
@@ -0,0 +1,180 @@
+"""Torch Module for GatedGCN layer"""
+# pylint: disable= no-member, arguments-differ, invalid-name, cell-var-from-loop
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from .... import function as fn
+
+
+class GatedGCNConv(nn.Module):
+    r"""GatedGCN: Residual Gated Graph ConvNets
+        <https://arxiv.org/pdf/1711.07553v2.pdf>`
+
+    .. math::
+        e_{ij}^{l+1}=D^{l} h_{i}^{l}+E^{l} h_{j}^{l}+C^l e_{ij}^{l}
+
+        norm_{ij}=\Sigma_{j\in N_{i}} \sigma\left(e_{ij}^{l+1}\right)+\varepsilon
+
+        \hat{e}_{ij}^{l+1}=\sigma(e_{ij}^{l+1}) \div norm_{ij}
+
+        h_{i}^{l+1}=A^l h_{i}^{l}+\Sigma_{j \in N_{i}} \hat{e}_{ij}^{l+1} \odot B^l h_{j}^{l}
+
+    Parameters
+    ----------
+    in_feats : int
+        Input feature size; i.e, the number of dimensions of :math:`\mathbf{x}_i`.
+    edge_feats: int
+        Edge feature size; i.e., the number of dimensions of :math:\mathbf{e}_{j,i}`.
+    out_feats : int
+        Output feature size; i.e., the number of dimensions of :math:`h_i^{(t+1)}`.
+    dropout : float, optional
+        Dropout rate on node and edge feature. Defaults: ``0``.
+    batch_norm : bool
+        Whether to include batch normalization on node . Default: ``True``.
+    residual : bool
+        Whether to include residual connection . Default: ``True``.
+    activation : callable activation function/layer or None, optional.
+        If not None, applies an activation function to the updated node features.
+        Default: ``None``.
+
+    Example
+    -------
+    >>> import dgl
+    >>> import torch as th
+    >>> import torch.nn.functional as F
+    >>> from dgl.nn import GatedGCNConv
+    >>> num_nodes, num_edges = 8, 30
+    >>> graph = dgl.rand_graph(num_nodes,num_edges)
+    >>> node_feats = th.rand((num_nodes, 20))
+    >>> edge_feats = th.rand((num_edges, 12))
+    >>> gatedGCN = GatedGCNConv(input_feats=20,
+    ...                         edge_feats=12,
+    ...                         output_feats=20,
+    ...                         dropout=0.2,
+    ...                         batch_norm=True,
+    ...                         residual=True,
+    ...                         activation=F.relu)
```

Comment:
You probably need to update this if you decide to update the arguments above. Simplify this model initialization step and use default values as much as possible.